plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/conversions/RemoveWrongExtraModifiersConversion.kt

```diff
@@ -0,0 +1,25 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+
+package org.jetbrains.kotlin.nj2k.conversions
+
+import org.jetbrains.kotlin.nj2k.NewJ2kConverterContext
+import org.jetbrains.kotlin.nj2k.RecursiveConversion
+import org.jetbrains.kotlin.nj2k.tree.*
+
+internal class RemoveWrongExtraModifiersConversion(context: NewJ2kConverterContext) : RecursiveConversion(context) {
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element !is JKOtherModifiersOwner) return recurse(element)
+        val objectClassParentExists = element.parents().none { parent -> parent is JKClass && parent.classKind == JKClass.ClassKind.OBJECT }
```

Comment:
Looks like it should be "doesn't exist" instead of "exists". I will clean this up myself along with some other things possibly, to speed up the review.

Reply:
AFAIU objects always can't be `inner`, so additional check of parents is redundant, I will fix this.